src/Sarif/Map/JsonMapBuilder.cs

```diff
@@ -15,58 +15,46 @@ namespace Microsoft.CodeAnalysis.Sarif.Map
     ///  JsonMapBuilder constructs a JsonMap for input Json files. It is passed
     ///  a size percentage target which controls how much detail it includes.
     /// </summary>
-    public class JsonMapBuilder
+    public static class JsonMapBuilder
     {
-        private const long NodeSizeEstimateBytes = 90;         // Bytes for one Node with containing property but not including children
-        private const long ArrayStartSizeEstimateBytes = 5;    // Bytes for one ArrayStart location, if under 10KB.
-
-        public double MaxFileSizePercentage { get; private set; }
-        public int MinimumSizeForNode { get; private set; }
-
-        /// <summary>
-        ///  Construct a JsonMapBuilder for the given target output size.
-        ///  (ex: 0.01 means 1% of the input file size)
-        /// </summary>
-        /// <param name="maxFileSizePercentage">Maximum size of the map relative to the source file (0.10 means 10%)</param>
-        public JsonMapBuilder(double maxFileSizePercentage)
-        {
-            if (maxFileSizePercentage <= 0 || maxFileSizePercentage >= 100) { throw new ArgumentOutOfRangeException("maxFileSizePercentage must be between 0 and 100."); }
-            MaxFileSizePercentage = maxFileSizePercentage;
-            MinimumSizeForNode = (int)(NodeSizeEstimateBytes / maxFileSizePercentage);
-        }
-
         /// <summary>
         ///  Build the JsonMap for a given Json file, by path.
         ///  Returns null if the source file was too small for any map nodes to fit the size budget.
         /// </summary>
         /// <param name="filePath">File Path to the Json file to build a map from</param>
+        /// <param name="settings">JsonMapSettings for map; null for defaults</param>
         /// <returns>JsonMap for file or null if file too small for map</returns>
-        public JsonMapNode Build(string filePath)
+        public static JsonMapNode Build(string filePath, JsonMapSettings settings = null)
         {
-            return Build(() => File.OpenRead(filePath));
+            return Build(() => File.OpenRead(filePath), settings);
         }
 
         /// <summary>
         ///  Build the JsonMap for a given Json file, given a stream provider.
         ///  Returns null if the source file was too small for any map nodes to fit the size budget.
         /// </summary>
         /// <param name="streamProvider">A function which will open the stream for the desired file</param>
+        /// <param name="settings">JsonMapSettings for map; null for defaults</param>
         /// <returns>JsonMap for file or null if file too small for map</returns>
-        public JsonMapNode Build(Func<Stream> streamProvider)
+        public static JsonMapNode Build(Func<Stream> streamProvider, JsonMapSettings settings = null)
```

Comment:
>Func<Stream> streamProvider [](start = 40, length = 27)

I've never seen this pattern before. Why does `Build` not just accept a `Stream`?

Reply:
Short version: 
- Deferred loading requires a way to build another copy of the stream (to go back and read a DeferredList just-in-time). 
- JsonPositionedTextReader must take this "streamProvider" in order to ferry it from the calling code through to the DeferredList
- JsonPositionedTextReader is required to identify byte offsets in the file for the map.

So, it's not needed for this specific scenario, but JsonPositionedTextReader is needed for map construction and deferred loading, and I don't want you to be able to create one that won't work for your scenario, so I ask for it unconditionally.